### PR TITLE
[full-ci][tests-only] Run e2e tests on keycloak

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR runs tests on keycloak in CI. For now, tests under journey are running on keycloak. 
`Journey` folder contains multiple user login and logout, sharing. 
On stable-8.0, all feature files mentioned below passes but on master spaces.feature has some problem so only journey is running for now
- tests/e2e/cucumber/features/smoke/admin-settings/users.feature:20 
- tests/e2e/cucumber/features/smoke/admin-settings/spaces.feature
- tests/e2e/cucumber/features/journeys
 master pr link:https://github.com/owncloud/web/pull/10481

## Related Issue

- Fixes <https://github.com/owncloud/web/issues/10458>

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

**Review by saw-jan — COMMENTED** [review]
Make this a main PR. This should be the first to get merged.
For now, let's run only `kindergarten.feature`
See https://github.com/owncloud/web/issues/10259#issuecomment-1978402015

File: .drone.star [hunk 1]
```diff
@@ -2168,3 +2183,154 @@ def appProviderService(name):
...
+            "ref": [
+                "refs/heads/master",
+                "refs/heads/stable-*",
+                "refs/tags/**",
+                "refs/pull/**",
+            ],
+            "status": [
+                "success",
+                "failure",
+            ],
```
Comment (saw-jan) [inline .drone.star:2334]: this trigger should not be needed, right?
```suggestion
```

File: tests/e2e/support/objects/app-admin-settings/users/index.ts [hunk 2]
```diff
@@ -9,17 +11,22 @@ export class Users {
     this.#usersEnvironment = new UsersEnvironment()
     this.#page = page
   }
-  getUUID({ key }: { key: string }): string {
-    return this.#usersEnvironment.getCreatedUser({ key }).uuid
+  async getUUID({ key }: { key: string }): Promise<string> {
+    if (config.keycloak) {
+      const user = this.#usersEnvironment.getUser({ key })
+      const admin = this.#usersEnvironment.getUser({ key: 'admin' })
+      return await getUserId({ user, admin })
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-admin-settings/users/index.ts:18]: will the id be same for the user throughout the single scenario? if yes then maybe a separate use store for keycloak and store respective ids when the user is created?

**Review by ScharfViktor — APPROVED** [review]
LGTM, thank you 👍

File: tests/e2e/support/objects/app-admin-settings/users/index.ts (hunk 2 above)
Comment (amrita-shrestha) [reply tests/e2e/support/objects/app-admin-settings/users/index.ts:18]: lets do that in separate PR

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**